tasks/emoji-unicode-version-map/index.js

```diff
@@ -1,119 +1,88 @@
+
+		const unicodeVersion = emoji.unicode_version.toFixed(1);
+
+		if (emoji.unicode_version !== parseFloat(unicodeVersion)) {
+			throw new Error(`Unsupported Unicode version ${emoji.unicode_version}`);
+		}
 
-	// Save the transformed emoji map
-	const fillingZwjSequencesPromise = Promise.all(addingEmojipediaUnicodeVersionPromises)
-		.then(() => {
-			console.log('Filling out ZWJ sequences...');
-			fillZwjSequenceUnicodeVersions(emojiMap);
+		emojiToUnicodeVersionMap[emojiNameKey] = unicodeVersion;
+
+		// Also loop through the aliases
+		(emoji.shortname_alternates || []).forEach((aliasShortName) => {
+			const aliasName = aliasShortName.slice(1, aliasShortName.length - 1);
+			emojiToUnicodeVersionMap[aliasName] = unicodeVersion;
 		});
+	});
 
 	// Setup the legacy aliases
-	const addingAliasesPromise = fillingZwjSequencesPromise
-		.then(() => {
-			Object.keys(aliasMap).forEach((aliasName) => {
-				const aliasReferenceName = aliasMap[aliasName];
-				emojiMap[aliasName] = Object.assign({}, emojiMap[aliasReferenceName]);
-			});
-		});
+	Object.keys(aliasMap).forEach((aliasName) => {
+		const aliasReferenceName = aliasMap[aliasName];
+		emojiToUnicodeVersionMap[aliasName] = emojiToUnicodeVersionMap[aliasReferenceName];
+	});
 
-	const ensureAllPopulatedPromise = addingAliasesPromise
-		.then(() => {
-			// Verify each emoji has a `unicodeVersion`
-			let missingVersionCount = 0;
-			let notStringVersionCount = 0;
-			Object.keys(emojiMap).forEach((emojiNameKey) => {
-				const emoji = emojiMap[emojiNameKey];
-				if(emoji.unicodeVersion === undefined) {
-					console.log(`Missing unicodeVersion property on \`${emojiNameKey}\``);
-					missingVersionCount += 1;
-				}
-				if(typeof emoji.unicodeVersion !== 'string') {
-					console.log(`Invalid unicodeVersion property, must be string on \`${emojiNameKey}\``);
-					notStringVersionCount += 1;
-				}
-			});
-			if(missingVersionCount > 0) {
-				throw new Error(`Missing unicodeVersion property on ${missingVersionCount} emojis`);
-			}
-			if(notStringVersionCount > 0) {
-				throw new Error(`unicodeVersion property not a string for ${notStringVersionCount} emojis`);
-			}
-		});
+	let missingVersionCount = 0;
+	let notStringVersionCount = 0;
 
+	Object.keys(emojiToUnicodeVersionMap).forEach((emojiNameKey) => {
+		const version = emojiToUnicodeVersionMap[emojiNameKey];
 
-	return ensureAllPopulatedPromise.then(() => {
-		const emojiToUnicodeVersionMap = {};
-		Object.keys(emojiMap).forEach((emojiNameKey) => {
-			const emojiInfo = emojiMap[emojiNameKey];
-			emojiToUnicodeVersionMap[emojiNameKey] = emojiInfo.unicodeVersion;
-			// Also loop through the aliases
-			(emojiInfo.aliases || []).forEach((aliasShortName) => {
-				const aliasName = aliasShortName.slice(1, aliasShortName.length - 1);
-				emojiToUnicodeVersionMap[aliasName] = emojiInfo.unicodeVersion;
-			});
-		});
-		return emojiToUnicodeVersionMap;
+		if(version === undefined) {
```

Comment:
Instead of checking for `undefined`, probably a bit safer to just go with `if(!version)`. We can also do this in `tasks/compare-maps.js`

Reply:
Since we had this before, let's just go with it 🚀